include/swift/Runtime/RuntimeFunctions.def

```diff
@@ -2273,12 +2273,12 @@ FUNCTION(TaskGroupDestroy,
          ATTRS(NoUnwind),
          EFFECT(Concurrency))
 
-// AutoDiffLinearMapContext *swift_autoDiffCreateLinearMapContext(size_t);
+// AutoDiffLinearMapContext *swift_autoDiffCreateLinearMapContext(const Metadata *);
```

Comment:
So, there will be no link errors for things compiled with old compiler and linked with new runtime and vice versa?

Reply:
Ah, it slipped my mind and I meant to ask this. I wasn't able to figure out how to make this a link time error.

Reply:
The easiest way: rename functions :) Hard way: invent some backward compatibility scheme. @rxwei any suggestions? I think we do not care about ABI stability for now, isn't it?

Reply:
Haven't made a decision on this in the latest revision. Waiting on @rxwei 's input. If breaking ABI is going to be a concern we could deprecate existing builtins instead of completely removing them.

Reply:
Should we deprecate the old ones as well? They do lead to memory leaks.

Reply:
What do you mean by deprecate? We just don't call it

Reply:
I mean, this is not an API. There is not a real way to deprecate it.

Reply:
@jkshtj I believe what @rxwei means is as follows:
 - Add new functions, say `swift_autoDiffCreateLinearMapContext2` etc. with the desired functionality
 - Keep old functions as-is

As a result: old code would still be able to run with the new runtime. The behavior will be the same, well, it will leak. The new code will call that new runtime functions and will be leak-less.